### PR TITLE
refactor graphql instrumentation for "NH" ...

### PR DESCRIPTION
**Pull request by prettycoder**
... because we wont have instrumentation residing in the graphql gem

- remove code that checks version and make sure the newer one is used
- remove corresponding tests and test files
- supported versions now 1.8.0 - 2.0.+

NH-10331

### CODE REVIEW

File: lib/solarwinds_apm/inst/graphql.rb [hunk 1]
```diff
@@ -4,247 +4,202 @@
...
-            kvs[:Key] = platform_key if (PREP_KEYS + EXEC_KEYS).include?(platform_key)
-
-            transaction_name(kvs[:InboundQuery]) if kvs[:InboundQuery] && layer == 'graphql.execute'
-
-            ::SolarWindsAPM::SDK.trace(layer, kvs: kvs) do
-              kvs.clear # we don't have to send them twice
-              yield
-            end
-          end
+        self.platform_keys = {
```
Comment (tammy-baylis-swi) [inline lib/solarwinds_apm/inst/graphql.rb:24]: Minor and a tangent: Is `self.platform_keys` used? Doing a grep for `platform_keys` within this repo doesn't give any results.

File: test/noop/noop_test.rb [hunk 2]
```diff
@@ -87,8 +87,7 @@ def test_padrino_not_instrumented
   end
 
   def test_graphql_not_instrumented
-    refute GraphQL::Schema.plugins.find { |plugin| plugin == GraphQL::Tracing::AppOpticsTracing },
-           'failed: This should be noop mode, but GraphQL is instrumented.'
+    refute defined?(GraphQL::Tracing::SolarWindsAPMTracing) && GraphQL::Schema.singleton_class.includes?(GraphQL::Tracing::SolarWindsAPMTracingrwindsAPM)
```
Comment (tammy-baylis-swi) [inline test/noop/noop_test.rb]: Minor: I think this is a typo: `SolarWindsAPMTracingrwindsAPM` --> `SolarWindsAPMTracing`

File: test/instrumentation/graphql_test.rb [hunk 3]
```diff
@@ -448,37 +444,21 @@ def company(id:)
...
+      size_1 = get_all_traces.size
+      clear_all_traces
+
+      SolarWindsAPMTest::Schema.use(GraphQL::Tracing::SolarWindsAPMTracing)
+      SolarWindsAPM::SDK.start_trace('graphql_test') do
+        SolarWindsAPMTest::Schema.execute(query_short)
+      end
+      size_2 = get_all_traces.size
+
+      assert_equal size_1, size_2
```
Comment (tammy-baylis-swi) [inline test/instrumentation/graphql_test.rb]: Trivial: Could we could keep the `'failed: duplicate plugins found'` message to resemble the original test? Ultimately it probably doesn't matter so we can leave it like this!

**Comment by tammy-baylis-swi** [comment]
Thanks @prettycoder ! A few very small suggestions from me; I'd say the typo is the only one out of what I've identified that should be fixed.

File: lib/solarwinds_apm/inst/graphql.rb (hunk 1 above)
Comment (prettycoder) [reply lib/solarwinds_apm/inst/graphql.rb:24]: It is part of the graphql api

File: test/noop/noop_test.rb (hunk 2 above)
Comment (prettycoder) [reply test/noop/noop_test.rb]: oops

File: test/instrumentation/graphql_test.rb (hunk 3 above)
Comment (prettycoder) [reply test/instrumentation/graphql_test.rb]: good point

File: lib/solarwinds_apm/inst/graphql.rb (hunk 1 above)
Comment (tammy-baylis-swi) [reply lib/solarwinds_apm/inst/graphql.rb:24]: Ahhh ok thanks!

**Review by tammy-baylis-swi — APPROVED** [review]
Lgtm! Thanks for the updates and passing tests

**Pull request merged**